### PR TITLE
Hmis 207 sonarcloud test

### PR DESCRIPTION
**Pull request by srinivasm5**
### Change description: Added yaml script to build pipeline for Sonarcloud test

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 trigger:
-  - master
+  - none
```
Comment (karanotts) [inline azure-pipelines.yml:3]: why are you changing trigger for the main pipeline?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -90,7 +90,10 @@ stages:
     ${{ if environment.condition }}:
       condition: ${{ environment.condition }}
     displayName: Build and Test ${{ environment.name }} APIM
-    
+    variables:
+    - group: SonarCloud
```
Comment (karanotts) [inline azure-pipelines.yml:94]: don't need to create another variable group, this can be placed within COMMON group

File: azure-pipelines.yml [hunk 3]
```diff
@@ -90,7 +90,10 @@ stages:
...
+    - name: Sonarcloud_login
```
Comment (karanotts) [inline azure-pipelines.yml:95]: no need to declare library variables

File: azure-pipelines.yml [hunk 4]
```diff
@@ -159,3 +162,12 @@ stages:
...
               storageContainer: $[dependencies.createStorage.outputs['createContainer.container']]
+    
+    - ${{ if ne(environment.name, 'dev') }}:
+      - template: pipeline/jobs/sonarqubetest.yaml
+        parameters:
+          displayName: 'Sonarcloudtest'
+          dependsOn:
+          - Terraform
+          - createStorage
+          command: sonarqube --stacktrace -$(Sonarcloud_login)
```
Comment (karanotts) [inline azure-pipelines.yml:173]: use library variable, no need to reassign it

File: pipeline/steps/run-sonarcloudanalysis.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+steps:
+- task: SonarSource.sonarcloud.14d9cde6-c1da-4d55-aa01-2965cd301255.SonarCloudPrepare@1
+  displayName: 'Prepare analysis on SonarCloud'
+  inputs:
+    SonarCloud: SonarCloud
+    organization: hmcts
+    scannerMode: Other
+
+- task: Gradle@2
```
Comment (karanotts) [inline pipeline/steps/run-sonarcloudanalysis.yaml:13]: this task already exists https://github.com/hmcts/hmi-api-gateway-fh/blob/e0e3f99e78543f1636ca4ff5ba0361f77450cc6d/pipeline/steps/run-gradlew.yaml

please use existing modules, we try to break down to single tasks so we don't need to repeat the code

File: azure-pipelines.yml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ---
 trigger:
-  - master
```
Comment (pats-john) [inline azure-pipelines.yml:3]: Why trigger set to none...Please change it back to master....

**Review by pats-john — CHANGES_REQUESTED** [review]
Further review should be given to Kara as she is much more of a Terraform expert than me...

**Review by karanotts — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml (hunk 6 above)
Comment (srinivasm5) [reply azure-pipelines.yml]: changed to master

File: azure-pipelines.yml (hunk 1 above)
Comment (srinivasm5) [reply azure-pipelines.yml]: changed to master

File: azure-pipelines.yml (hunk 2 above)
Comment (srinivasm5) [reply azure-pipelines.yml]: I have added new variable in common group

File: pipeline/steps/run-sonarcloudanalysis.yaml (hunk 5 above)
Comment (srinivasm5) [reply pipeline/steps/run-sonarcloudanalysis.yaml]: Used existing Gradle tasks into sonarcloud.

File: azure-pipelines.yml (hunk 4 above)
Comment (srinivasm5) [reply azure-pipelines.yml]: used library variable and removed reassign variable

**Review by karanotts — APPROVED** [review]

**Comment by srinivasm5** [comment]
All checks passed.

**Pull request merged**